prodigy_pdf/spans.py

```diff
@@ -0,0 +1,304 @@
+                    if not self.hide_preview:
+                        blocks.append({"view_id": "image", "spans": image_spans})
+                    eg = {
+                        "text": span.text,
+                        "image": images[i],
+                        "tokens": get_layout_tokens(span, disabled=disabled),
+                        "width": page_layout.width,
+                        "height": page_layout.height,
+                        "view_id": "blocks",
+                        "config": {"blocks": blocks},
+                        "text_span": {
+                            "token_start": span.start,
+                            "token_end": span.end - 1,
+                            "start": span.start_char,
+                            "end": span.end_char,
+                            "text": span.text,
+                            "label": span.label_,
+                        },
+                        "meta": {"title": file_path.stem, "page": page_layout.page_no},
+                    }
+                    yield set_hashes(eg)
+
+
+@recipe(
+    "pdf.spans.manual",
+    # fmt: off
+    dataset=Arg(help="Dataset to save annotations to"),
+    nlp=Arg(help="Loadable spaCy pipeline"),
+    source=Arg(help="Path to directory to load from"),
+    labels=Arg("--label", "-l", help="Comma-separated label(s) to annotate or text file with one label per line"),
+    add_ents=Arg("--add-ents", "-E", help="Add named enitites for the given labels via the spaCy model"),
+    focus=Arg("--focus", "-f", help="Focus mode: annotate selected sections of a given type, e.g. 'text'"),
+    disable=Arg("--disable", "-d", help="Labels of layout spans to disable, e.g. 'footnote'"),
+    split_pages=Arg("--split-pages", "-S", help="View pages as separate tasks"),
+    hide_preview=Arg("--hide-preview", "-HP", help="Hide side-by-side preview of layout"),
+    # fmt: on
+)
+def pdf_spans_manual(
+    dataset: str,
+    nlp: Language,
+    source: str,
+    labels: Optional[List[str]] = None,
+    add_ents: bool = False,
+    focus: Optional[List[str]] = None,
+    disable: Optional[List[str]] = None,
+    hide_preview: bool = False,
+    split_pages: bool = False,
+) -> ControllerComponentsDict:
+    """
+    Apply span annotations to text-based document contents extracted with
+    spacy-layout and Docling. For efficiency, the recipe can run with
+    --focus text to walk through individual text blocks, which are highlighted
+    in a visual preview of the document page.
+    """
+    log("RECIPE: Starting recipe pdf.spans.manual", locals())
+    view_id = "spans_manual"
+    layout_stream = LayoutStream(
+        source,
+        nlp=nlp,
+        file_ext=["pdf"],
+        view_id=view_id,
+        disable=disable or [],
+        split_pages=split_pages,
+        hide_preview=hide_preview,
+        focus=focus or [],
+    )
+    stream = Stream.from_iterable(layout_stream.get_stream())
+    if add_ents:
```

Comment:
```suggestion
    if add_ents:
         labels = resolve_labels(nlp, component, recipe_labels=labels)
```
Otherwise if the user specifies --add-ents and does not provide labels (because they want all model labels) nothing gets added as labels is None.